src/gcp/cloudfunctionsv2.ts

```diff
@@ -436,12 +435,27 @@ export function functionFromEndpoint(endpoint: backend.Endpoint, source: Storage
       eventType: endpoint.eventTrigger.eventType,
     };
     if (gcfFunction.eventTrigger.eventType === PUBSUB_PUBLISH_EVENT) {
-      gcfFunction.eventTrigger.pubsubTopic = endpoint.eventTrigger.eventFilters.resource;
-    } else {
       gcfFunction.eventTrigger.eventFilters = [];
-      for (const [attribute, value] of Object.entries(endpoint.eventTrigger.eventFilters)) {
-        gcfFunction.eventTrigger.eventFilters.push({ attribute, value });
+
+      const pubsubFilter = backend.findEventFilter(endpoint, "topic");
+      if (!pubsubFilter) {
+        throw new FirebaseError(
+          "Invalid pubsub endpoint. Expected eventFilter with 'topic' attribute but found none."
+        );
       }
+      gcfFunction.eventTrigger.pubsubTopic = pubsubFilter.value;
+
+      for (const filter of endpoint.eventTrigger.eventFilters) {
+        if (filter.attribute === "topic") {
+          continue;
+        }
+        if (!gcfFunction.eventTrigger.eventFilters) {
+          gcfFunction.eventTrigger.eventFilters = [];
+        }
+        gcfFunction.eventTrigger.eventFilters.push(filter);
+      }
```

Comment:
I think we could just inline this whole loop to 
```
gcfFunction.eventTrigger.eventFilters = endpoint.eventTrigger.eventFilters.filter((ef) => ef.attribute !== "topic"); 
```

Reply:
See comment above. I did start off with this but then realized that

1. `eventTrigger.eventFilters = []`
2. `eventTrigger.eventFilters = undefined`

Are different things when it comes to deployment - the former would override whatever eventFilter values that changed out-of-band while the later preserves it.

Reply:
Ahhh that makes sense, thanks